File: cp_nim/src/bin/abc404-a-not-found.nim
# Created by Ayush Biswas at 2026/07/18 17:50
# https://atcoder.jp/contests/adt_easy_20260625_1/tasks/abc404_a
import ../lib/cpio

# @code begin

import sequtils

var r = newReader()

let s = r[string]

for c in 'a'..'z':
  if not s.contains(c):
    echo c
    break

# @code end

File: cp_rust/src/bin/abc404-a-not-found.rs
// Created by Ayush Biswas at 2026/07/06 22:54
// https://atcoder.jp/contests/adt_easy_20260625_1/tasks/abc404_a
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        s: [char]
    ) -> char {
        let mut z = s.into_iter().sorted().group_by(|&x| x).map(|g| g[0]);
        for a in 'a'..='z' {
            if let Some(b) = z.next() {
                if a != b {
                    return a;
                }
            } else {
                return a;
            }
        }
        'z'
    }
}
// @code end
